crates/nu-cli/src/repl.rs

```diff
@@ -418,6 +485,16 @@ pub fn evaluate_repl(
         start_time = std::time::Instant::now();
         line_editor = line_editor.with_transient_prompt(transient_prompt);
         let input = line_editor.read_line(&nu_prompt);
+        // we got our inputs, we can now drop our stack references
+        // CLEAR STACK-REFERENCE 1
+        line_editor = line_editor
+            .with_highlighter(Box::new(SimpleMatchHighlighter::new("hi".to_string())))
+            // CLEAR STACK-REFERENCE 2
+            .with_completer(Box::<DefaultCompleter>::default())
+            // CLEAR STACK-REFERENCE 3
+            .clear_menus()
+            // CLEAR STACK-REFERENCE 4
+            .with_transient_prompt(Box::<DefaultPrompt>::default());
```

Comment:
this clears references back to `stack` in the child stacks, letting us mutate through the original `Arc`, and avoid any copies.